### PR TITLE
#913 - Deprecated the gender column

### PR DESCRIPTION
**Pull request by strell**
### Description
 - Added new field genderV2 with the type of enum
 - Added data mapper in order to turn the gender value into genderV2 value and vice-versa
 - Extended api-functional tests

### Fixed Issues
1. magento/graphql-ce#913: [Customer] Convert gender type to enum

### Manual testing scenarios
See issue description

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/CustomerGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -63,7 +68,8 @@ input CustomerInput {
     email: String @doc(description: "The customer's email address. Required")
     dob: String @doc(description: "The customer's date of birth")
     taxvat: String @doc(description: "The customer's Tax/VAT number (for corporate customers)")
-    gender: Int @doc(description: "The customer's gender(Male - 1, Female - 2)")
+    gender: Int @deprecated(reason: "Deprecated, use the genderV2 instead") @doc(description: "The customer's gender(Male - 1, Female - 2)")
+    genderV2: CustomerGenderEnum @doc(description: "The customer's gender")
```
Comment (lenaorobei) [inline app/code/Magento/CustomerGraphQl/etc/schema.graphqls:72]:
```suggestion
    gender_v2: CustomerGenderEnum @doc(description: "The customer's gender")
```

File: app/code/Magento/CustomerGraphQl/etc/schema.graphqls [hunk 2]
```diff
@@ -92,7 +98,8 @@ type Customer @doc(description: "Customer defines the customer name and address
     id: Int @doc(description: "The ID assigned to the customer")
     is_subscribed: Boolean @doc(description: "Indicates whether the customer is subscribed to the company's newsletter") @resolver(class: "\\Magento\\CustomerGraphQl\\Model\\Resolver\\IsSubscribed")
     addresses: [CustomerAddress] @doc(description: "An array containing the customer's shipping and billing addresses") @resolver(class: "\\Magento\\CustomerGraphQl\\Model\\Resolver\\CustomerAddresses")
-    gender: Int @doc(description: "The customer's gender(Male - 1, Female - 2)")
+    gender: Int @deprecated(reason: "Deprecated, use the genderV2 instead") @doc(description: "The customer's gender(Male - 1, Female - 2)")
+    genderV2: CustomerGenderEnum @doc(description: "The customer's gender")
```
Comment (lenaorobei) [inline app/code/Magento/CustomerGraphQl/etc/schema.graphqls:102]:
```suggestion
    gender_v2: CustomerGenderEnum @doc(description: "The customer's gender")
```

File: app/code/Magento/CustomerGraphQl/etc/graphql/di.xml [hunk 3]
```diff
@@ -20,4 +20,60 @@
             </argument>
         </arguments>
     </type>
+    <preference for="Magento\CustomerGraphQl\Api\DataMapperInterface" type="Magento\CustomerGraphQl\Model\Customer\DataMapper\DataMapperPool"/>
```
Comment (lenaorobei) [inline app/code/Magento/CustomerGraphQl/etc/graphql/di.xml:23]: Some logic here is redundant. Please have a look at `Magento/BundleGraphQl/etc/graphql/di.xml:27` as example.

File: dev/tests/integration/testsuite/Magento/Customer/_files/customer.php [hunk 4]
```diff
@@ -32,3 +32,27 @@
 $customer->isObjectNew(true);
 $customer->save();
 $customerRegistry->remove($customer->getId());
+
```
Comment (lenaorobei) [inline dev/tests/integration/testsuite/Magento/Customer/_files/customer.php:35]: This file is used in many other tests. If you need some specific customer behaviour, please create new fixture.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please address review comments.

**Comment by strell** [comment]
@lenaorobei thank you for you cr, yeah, I just have found out that Magento has such kind of mapper under Magento\Framework\GraphQl\ already, so I will refactor my code using this mapper.

**Comment by fooman** [comment]
It might not be the best place for this discussion but I feel using an Enum here for the list of genders makes us less inclusive. Example article here https://www.itspronouncedmetrosexual.com/2012/06/how-can-i-make-the-gender-question-on-an-application-form-more-inclusive/ 

I would generally consider best practice (if you even have to ask) to include a freeform entry. Switching to Enum would not allow for this. But still maybe a good discussion to have before needing to do gender_v3.

**Comment by lenaorobei** [comment]
Closing this PR due to inactivity. @strell please feel free to reassign the issue and reopen this PR if you are interested to continue working on it.
Thank you.

**Pull request closed without merging**